### ISSUE TITLE
jenkins-slave don't stop correctly

### ISSUE THREAD

**Issue by xdexter** [jenkinsci/puppet-jenkins#557]
Hello,

I'm using Centos 6.7 and noticed a problem during the process of shutdown, now when the node slave is shut down due to the LOCK path the operating system does not effect the stop correctly, this creates a problem in Jenkins Master where the node slave is not removed.

To correct simply change the file "jenkins-slave.RedHat":
- LOCK_FILE = /var/lock/$NAME
+ LOCK_FILE = /var/lock/subsys/$NAME

You can this fix be analyzed?

**Comment by jhoblitt** [comment]
@xdexter That is unusual and not the behavior that I'm seeing:

https://github.com/jenkinsci/puppet-jenkins/blob/master/files/jenkins-slave.RedHat#L13

```
[vagrant@jenkins-el6-1 ~]$ cat /etc/redhat-release 
CentOS release 6.7 (Final)
[vagrant@jenkins-el6-1 ~]$ ls -la /var/lock/jenkins-slave /var/lock/subsys/jenkins-slave
ls: cannot access /var/lock/subsys/jenkins-slave: No such file or directory
-rw-r--r-- 1 root root 0 Mar 15 11:32 /var/lock/jenkins-slave
```

What is recreating `/var/lock/subsys/jenkins-slave`?

**Comment by xdexter** [comment]
Because when there is a lock file "/var/lock/subsys/jenkins-slave" operating system will wait for the end of the process, and it works very well. As in this case the file is not in this directory, the stop does not work properly, if you create a Master / Slave setting will see that when the slave node is off the node remains in Jenkins Master, understand me?

**Comment by jhoblitt** [comment]
@xdexter I believe I understand now -- the init script does stop the slave properly but the sysvinit system on centos 6 is not waiting for the service shutdown because the lock file is not in the expected place.

**Comment by xdexter** [comment]
@jhoblitt right, that's it. Changing the path as above inform it works properly.

**Comment by xdexter** [comment]
@jhoblitt. How is the procedure now to fix? Is there anything else I can help?

**Comment by jhoblitt** [comment]
@xdexter The development process for this module is fairly informal. All that is required is for the fix to submitted as a PR.

**Comment by xdexter** [comment]
@jhoblitt I apologize but I did not understand.

**Comment by jhoblitt** [comment]
@xdexter PR is an abbreviation for "Pull Request".  If you are not familiar with the term, my suggestion would be to google for "github pull request".  There are numerous tutorials that walk through the process of submitting a pull request.

**Comment by xdexter** [comment]
@jhoblitt Right, would it? https://github.com/jenkinsci/puppet-jenkins/pull/562

**Comment by jhoblitt** [comment]
Resolved via #562
